Components/Transforms/AdvancedLimitedEulerTransform/elxAdvancedLimitedEulerTransform.hxx

```diff
@@ -0,0 +1,588 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef __elxAdvancedLimitedEulerTransform_HXX_
+#define __elxAdvancedLimitedEulerTransform_HXX_
+
+#include "elxAdvancedLimitedEulerTransform.h"
+#include "itkImageGridSampler.h"
+#include "itkContinuousIndex.h"
+
+namespace elastix
+{
+
+/**
+ * ********************* Constructor ****************************
+ */
+
+template< class TElastix >
+AdvancedLimitedEulerTransformElastix< TElastix >
+::AdvancedLimitedEulerTransformElastix()
+{
+  this->m_LimitedEulerTransform = LimitedEulerTransformType::New();
+  this->SetCurrentTransform( this->m_LimitedEulerTransform );
+
+} // end Constructor
+
+
+/**
+ * ******************* BeforeRegistration ***********************
+ */
+
+template< class TElastix >
+void
+AdvancedLimitedEulerTransformElastix< TElastix >
+::BeforeRegistration( void )
+{
+  /** Set center of rotation and initial translation. */
+  this->InitializeTransform();
+
+  /** Set the scales. */
+  this->m_LimitedEulerTransform->SetScalesEstimation(true);
+  this->SetScales();
+  this->m_LimitedEulerTransform->SetScalesEstimation(false);
+
+} // end BeforeRegistration()
+
+
+/**
+ * ************************* ReadFromFile ************************
+ */
+
+template< class TElastix >
+void
+AdvancedLimitedEulerTransformElastix< TElastix >
+::ReadFromFile( void )
+{
+  /** Variables. */
+  InputPointType centerOfRotationPoint;
+  centerOfRotationPoint.Fill( 0.0 );
+
+  /** Try first to read the CenterOfRotationPoint from the
+   * transform parameter file, this is the new, and preferred
+   * way, since elastix 3.402.
+   */
+  bool pointRead = this->ReadCenterOfRotationPoint( centerOfRotationPoint );
+
+  if( !pointRead )
+  {
+    xl::xout[ "error" ] << "ERROR: No center of rotation is specified in "
+                        << "the transform parameter file" << std::endl;
+    itkExceptionMacro( << "Transform parameter file is corrupt." )
+  }
+
+  /** Set the center in this Transform. */
+  this->m_LimitedEulerTransform->SetCenter( centerOfRotationPoint );
+
+  /** Read the ComputeZYX. */
+  if( SpaceDimension == 3 )
+  {
+    std::string computeZYX = "false";
+    this->m_Configuration->ReadParameter( computeZYX, "ComputeZYX", 0 );
+    if( computeZYX == "true" )
+    {
+      this->m_LimitedEulerTransform->SetComputeZYX( true );
+    }
+  }
+
+  /** Read SharpnessOfLimits parameter */
+  ScalarType sharpnessOfLimits = 50.0f;
+  this->m_Configuration->ReadParameter( sharpnessOfLimits, "SharpnessOfLimits", 0 );
+  this->m_LimitedEulerTransform->SetSharpnessOfLimits( sharpnessOfLimits );
+
+  /** Read and set UpperLimits parameter */
+  ParametersType upperLimits;
+  upperLimits.SetSize(6);
+  for( unsigned int i = 0; i < 6; i++ )
+  {
+    upperLimits[ i ] = i < 3 ? 3.14f : 200.0f;
```

Comment:
Why did you specifically choose `200.0f` here? Would `std::numeric_limits<float>::max()` be fine as well, as a default? Or even `std::numeric_limits<float>::infinity()`?

If you still think `200.0f` is the most appropriate default value, can you please make it a named constant? Possibly a `static constexpr` member of the class?